src/node_process_methods.cc

```diff
@@ -160,11 +160,25 @@ static void Kill(const FunctionCallbackInfo<Value>& args) {
 static void MemoryUsage(const FunctionCallbackInfo<Value>& args) {
   Environment* env = Environment::GetCurrent(args);
 
+  // Get the double array pointer from the Float64Array argument.
+  CHECK(args[0]->IsFloat64Array());
+  Local<Float64Array> array = args[0].As<Float64Array>();
+  CHECK_EQ(array->Length(), 5);
+  Local<ArrayBuffer> ab = array->Buffer();
+  double* fields = static_cast<double*>(ab->GetBackingStore()->Data());
+
   size_t rss;
   int err = uv_resident_set_memory(&rss);
   if (err)
     return env->ThrowUVException(err, "uv_resident_set_memory");
 
+  fields[0] = rss;
+
+  Local<v8::Boolean> rssOnly = args[1].As<v8::Boolean>();
+  if (rssOnly->Value()) {
+    return;
+  }
```

Comment:
```suggestion
  if (args[1]->IsTrue()) return;
```

Reply:
I had declared an intermediary variable for sack of readability but I'm not very familiar with Node guideline about this

Reply:
As a rule of thumb: shorter is better. :-)